File: src/types/Event.ts
export interface Event {
  id: string
  userId: string
  name: string
  title?: string
  startDate: string
  startTime?: string
  endDate?: string
  endTime?: string
  type: 'online' | 'offline'
  privacy: 'public' | 'private' | 'group'
  description?: string
  images: string[]
  location?: string
  locationName?: string
  address?: string
  showGuestList?: boolean
  hosts?: string[]
  date?: string
  responses?: number
  guestsGoing?: number
  guestsInterested?: number
  coordinates?: number[]
  frequency?: string
}

File: backend/search-service/src/main/java/com/facebook/SearchService/model/Event.java
package com.facebook.SearchService.model;

import lombok.Data;

@Data
public class Event {
    private String id;
    private String userId;
    private String name;
    private String title;
    private String startDate;
    private String startTime;
    private String endDate;
    private String endTime;
    private String type;
    private String privacy;
    private String description;
    private String location;
    private String locationName;
    private String address;
    private Boolean showGuestList;
    private String date;
    private String frequency;
}
